src/utils/special-paths.js

```diff
@@ -275,16 +276,35 @@ export function getDownloadRecipeForSourceFile(
       };
     }
     case 'gem': {
-      const { gem, path } = parsedFile;
-      return {
-        type: 'CORS_ENABLED_SINGLE_FILE',
-        url: `https://gems.vernier.prof/${gem}/${path}`
-      };
+      if (typeof rubyGemDownloadRecipe === 'function') {
+        return rubyGemDownloadRecipe(parsedFile);
```

Comment:
Note that it is up to the implementer of these hooks to ensure that they receive the right param type, and return the right return type, since we cannot import the types in this file without a circular reference issue.